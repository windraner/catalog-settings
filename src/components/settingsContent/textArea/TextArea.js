import React from 'react';
import PropTypes from 'prop-types';

import './inputTextArea.css';

const TextArea = (props) => {
  return (
    <textarea
      value={props.value ? props.value : ''}
      onChange={(e) => props.onChange(props.name, e.target.value)}
      onBlur={(e) => props.onChange(props.name, e.target.value.trim())}
      type="text"
      className="catalog-settings-content__textarea"
    />
  );
};

TextArea.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
};

export default TextArea;
