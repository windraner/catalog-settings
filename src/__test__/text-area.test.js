import React from 'react';
import TextArea from '../components/settingsContent/textArea/text-area';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('<TextArea />', () => {
  it('Check for className and text', () => {
    const wrapper = shallow(<TextArea value="test" onChange={function() {}} />);
    expect(wrapper.find('.catalog-settings-content__textarea').length).toBe(1);
    expect(wrapper.find('.catalog-settings-content__textarea').props().value).toBe('test');
  });
  it('matches the snapshot', () => {
    const tree = shallow(<TextArea value="test" onChange={function() {}} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
