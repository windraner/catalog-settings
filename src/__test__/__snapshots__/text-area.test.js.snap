// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextArea /> matches the snapshot 1`] = `
<textarea
  className="catalog-settings-content__textarea d-block w-100 mt-1"
  onBlur={[Function]}
  onChange={[Function]}
  type="text"
  value="test"
/>
`;
